tests/manage/pv_services/test_expansion_snapshot_clone.py

```diff
@@ -0,0 +1,398 @@
+import logging
+import pytest
+
+from ocs_ci.helpers import helpers
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.resources import pod
+from ocs_ci.framework.testlib import (
+    ManageTest, tier2, skipif_ocs_version, skipif_ocp_version
+)
+
+log = logging.getLogger(__name__)
+
+
+@tier2
+@skipif_ocs_version('<4.6')
+@skipif_ocp_version('<4.6')
+@pytest.mark.polarion_id('OCS-2408')
+class TestExpansionSnapshotClone(ManageTest):
+    """
+    Tests to verify snapshot, clone and expansion
+
+    """
+    @pytest.fixture(autouse=True)
+    def setup(
+        self, project_factory, snapshot_restore_factory, pvc_clone_factory,
+        create_pvcs_and_pods
+    ):
+        """
+        Create PVCs and pods
+
+        """
+        self.pvc_size = 2
+        self.pvcs, self.pods = create_pvcs_and_pods(
+            pvc_size=self.pvc_size,
+            access_modes_rbd=[constants.ACCESS_MODE_RWO],
+            access_modes_cephfs=[constants.ACCESS_MODE_RWO]
+        )
+
+    def test_expansion_snapshot_clone(
+        self, snapshot_factory, snapshot_restore_factory, pvc_clone_factory,
+        pod_factory
+    ):
+        """
+        This test performs the following operations :
+
+        Expand parent PVC --> Take snapshot --> Expand parent PVC -->
+        Take clone --> Restore snapshot --> Expand cloned and restored PVC -->
+        Clone restored PVC --> Snapshot and restore of cloned PVCs -->
+        Expand new PVCs
+
+        Data integrity will be checked in each stage as required.
+        This test verifies that the clone, snapshot and parent PVCs are
+        independent and any operation in one will not impact the other.
+
+        """
+        filename = 'fio_file'
+        filename_restore_clone = 'fio_file_restore_clone'
+        pvc_size_expand_1 = 4
+        pvc_size_expand_2 = 6
+        pvc_size_expand_3 = 8
+        snapshots = []
+
+        # Run IO
+        log.info("Start IO on pods")
+        for pod_obj in self.pods:
+            log.info(f"Running IO on pod {pod_obj.name}")
+            pod_obj.run_io(
+                storage_type='fs', size='1G', runtime=20, fio_filename=filename
+            )
+        log.info("IO started on all pods")
+
+        log.info("Wait for IO completion on pods")
+        for pod_obj in self.pods:
+            pod_obj.get_fio_results()
+            log.info(f"IO finished on pod {pod_obj.name}")
+            # Calculate md5sum
+            md5sum = pod.cal_md5sum(pod_obj, filename)
+            pod_obj.pvc.md5sum = md5sum
+        log.info("IO completed on all pods")
```

Comment:
there are some other test cases under tests/manage/pv_services that implement the same.
The problem with that is if changes are needed, all implementations will need to be changed. 
For example, 
https://github.com/red-hat-storage/ocs-ci/blob/86ad959efefb89ab12ba9c16180636062c8c7d54/tests/manage/pv_services/pvc_snapshot/test_snapshot_at_different_pvc_utlilization_level.py#L65

Would it be possible to create a function that contains this code and call it from all places?

Reply:
@ebenahar Are you suggesting a function which accepts a list of pods and check IO status in each pod ?
Checking md5sum depends on test design. Here md5sum is obtained and added an attribute of PVC instance. This is done for verification in a later stage. If pod is not created using factory, pvc attribute will not be present in pod_obj. Some tests cases needs md5sum comparison after IO where the value to compare will be present is some list or dict. Other test cases like https://github.com/red-hat-storage/ocs-ci/blob/86ad959efefb89ab12ba9c16180636062c8c7d54/tests/manage/pv_services/pvc_snapshot/test_snapshot_at_different_pvc_utlilization_level.py#L65 will need to add the md5sum value in a dict.